### DISCUSSION TITLE
defaultValue could not be set for Checkbox controller component

### DISCUSSION THREAD

**Discussion by Adityamerc** [react-hook-form/react-hook-form#5427] [Q&A]
**Describe the bug**
I used the Checkbox from material-ui in the Controller component, the defaultValue prop is not working for this component. I've tried it in inline as well as in the useForm hook under the defaultValues key but it's not working.

**To Reproduce**
1. Initialize a form using react-hook-form
2. Import Checkbox component from @material-ui
3. Connect the component to the form using Controller from react-hook-form
4. Set a defaultValue, which must be a boolean, either directly in the props of the Controller component or inside the useForm hook
5. Now the set defaultValue would not be applied to the Checkbox component

**Codesandbox link (Required)**
https://codesandbox.io/s/elated-bird-oul5r?file=/src/Main.js

**Expected behavior**
When the defaultValue is set for a particular field it must be reflected. For example if I set the defaultValue to be true the Checkbox should be checked and vice versa

**Desktop (please complete the following information):**

- OS: Microsoft Windows 10 Enterprise
- Browser: Chrome
- Version: 90

**Additional context**
It'd really great if it was fixed asap, I know the people working on this are far busier than I am, but since a Checkbox is a ubiquitous in the dev community, this issue fixed asap would be so helpful as most of the devs love react-hook-form including me.

Thanks in advance

**Comment by thanh-nguyen-95** [comment]
You can use `<Checkbox {...field} checked={field.value} />` for now.

In my opinion, this issue is specific to an UI library and it should be solved by the user. In you case, it is `checked` prop but it could be `isChecked` or `someRandomPropNameBecauseICan` for me (a bit exaggerated but you get the point).

**Comment by Adityamerc** [comment]
But I need to load the the initialValue for that Checkbox, before I even begin to interact with it.

**Reply by thanh-nguyen-95** [reply]
It will reflect the initial value (set in `useForm({ defaultValues: { checkbox: true } })` on the checkbox if you do `<Checkbox {...field} checked={field.value} />`
